third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -475,11 +475,29 @@ class BlockedToMFMA : public OpRewritePattern<tt::DotOp> {
     // requires to broadcast the operand A.
     bool isTransposed = !(mDim == 4 && nDim == 64);
     auto aElemTy = mfmaInstr->aElementType;
-    ttg::AMDMfmaEncodingAttr mfmaEnc = ttg::AMDMfmaEncodingAttr::get(
-        oldRetType.getContext(),
-        /*version*/ mfmaVersion, warpsPerTile,
-        /*instrShape*/ mDim, nDim, /*isTransposed=*/isTransposed, CTALayout,
-        mfmaAccType);
+    bool isMfma16InBwdFA =
+        mfmaVersion == 4 && (aElemTy.isF16() || aElemTy.isBF16()) &&
+        mDim == 16 && nDim == 16 && oldAType.getRank() == 2 &&
+        oldAType.getShape().front() >= 16 * 2 &&
+        oldBType.getShape().back() == 16 && isTransposed &&
+        warpsPerTile.back() == 1;
+
+    ttg::AMDMfmaEncodingAttr mfmaEnc;
```

Comment:
We don't need to if-else the whole `mfmaEnc`; we can just define a local variable `tilesPerWarp = {isMfma16InBwdFA ? 2 : 1, 1}`?